lightning/src/blinded_path/mod.rs

```diff
@@ -162,3 +247,30 @@ impl_writeable!(BlindedHop, {
 	encrypted_payload
 });
 
+impl Direction {
+	/// Returns the [`NodeId`] from the inputs corresponding to the direction.
+	pub fn select_node_id<'a>(&self, node_a: &'a NodeId, node_b: &'a NodeId) -> &'a NodeId {
+		let (node_one, node_two) = if node_a < node_b {
+			(node_a, node_b)
+		} else {
+			(node_b, node_a)
+		};
+		match self {
+			Direction::NodeOne => node_one,
+			Direction::NodeTwo => node_two,
+		}
+	}
```

Comment:
I think we can simplify this function by using min, and max. Also, I think it will good idea to use them here considering the wording we have used in the `Direction` Enum.

```rust
pub enum Direction {
	/// The lesser node id when compared lexicographically in ascending order.
	NodeOne,
	/// The greater node id when compared lexicographically in ascending order.
	NodeTwo,
}
```

```suggestion
		match self {
			Direction::NodeOne => core::cmp::min(node_a, node_b),
			Direction::NodeTwo => core::cmp::max(node_a, node_b),
		}
```

Reply:
Nice!